src/libraries/System.Security.Cryptography/tests/MLDsaCngTests.Windows.cs

```diff
@@ -0,0 +1,92 @@
+
+        protected override MLDsaCng ImportSecretKey(MLDsaAlgorithm algorithm, ReadOnlySpan<byte> source) =>
+            MLDsaTestHelpers.ImportSecretKey(algorithm, source, CngExportPolicies.AllowExport);
+
+        protected override MLDsaCng ImportPublicKey(MLDsaAlgorithm algorithm, ReadOnlySpan<byte> source) =>
+            MLDsaTestHelpers.ImportPublicKey(algorithm, source);
+    }
+
+    [ConditionalClass(typeof(MLDsa), nameof(MLDsa.IsSupported))]
+    public sealed class MLDsaCngTests
+    {
+        [Theory]
+        [MemberData(nameof(MLDsaTestsData.IetfMLDsaAlgorithms), MemberType = typeof(MLDsaTestsData))]
+        public void ImportPrivateKey_NoExportFlag(MLDsaKeyInfo info)
+        {
+            using MLDsa mldsa = MLDsaTestHelpers.ImportSecretKey(info.Algorithm, info.SecretKey, CngExportPolicies.None);
+
+            MLDsaTestHelpers.AssertExportMLDsaPublicKey(export =>
+                AssertExtensions.SequenceEqual(info.PublicKey, export(mldsa)));
+
+            MLDsaTestHelpers.AssertExportMLDsaSecretKey(export =>
+                AssertExtensions.ThrowsContains<CryptographicException>(() => export(mldsa), "The requested operation is not supported"));
+
+            MLDsaTestHelpers.AssertExportMLDsaPrivateSeed(export =>
+                AssertExtensions.ThrowsContains<CryptographicException>(() => export(mldsa), "The requested operation is not supported"));
+        }
+
+        [Theory]
+        [MemberData(nameof(MLDsaTestsData.IetfMLDsaAlgorithms), MemberType = typeof(MLDsaTestsData))]
+        public void ImportPrivateSeed_NoExportFlag(MLDsaKeyInfo info)
+        {
+            using MLDsa mldsa = MLDsaTestHelpers.ImportPrivateSeed(info.Algorithm, info.PrivateSeed, CngExportPolicies.None);
+
+            MLDsaTestHelpers.AssertExportMLDsaPublicKey(export =>
+                AssertExtensions.SequenceEqual(info.PublicKey, export(mldsa)));
+
+            MLDsaTestHelpers.AssertExportMLDsaSecretKey(export =>
+                AssertExtensions.ThrowsContains<CryptographicException>(() => export(mldsa), "The requested operation is not supported"));
+
+            MLDsaTestHelpers.AssertExportMLDsaPrivateSeed(export =>
+                AssertExtensions.ThrowsContains<CryptographicException>(() => export(mldsa), "The requested operation is not supported"));
+        }
+
+        [Fact]
+        public void MLDsaCng_Ctor_ArgValidation()
+        {
+            AssertExtensions.Throws<ArgumentNullException>("key", static () => new MLDsaCng(null));
+        }
+
+        [Fact]
+        public void MLDsaCng_WrongAlgorithm()
+        {
+            using RSACng rsa = new RSACng();
+            using CngKey key = rsa.Key;
+            Assert.Throws<ArgumentException>("key", () => new MLDsaCng(key));
+        }
+
+        // TODO MLDsaCng doesn't have a public DuplicateHandle like OpenSSL since CngKey does that
+        // internally with CngKey.Handle. Is there something else we should test for copy/duplication?
```

Comment:
* Make an appropriate CngKey with no name.
* Make a bajunch of MLDsaCng instances from it (3? 20? whatever)
* Dispose all but one of them (ideally neither the first nor last instance created)
* Make sure the surviving one still works.

Repeating with a named key is not terrible.  But don't forget to delete the key.